app/feature/feature-terminate-insurance/src/main/kotlin/com/hedvig/android/feature/terminateinsurance/step/choose/ChooseInsuranceToTerminateViewModel.kt

```diff
@@ -0,0 +1,183 @@
+
+internal class ChooseInsuranceToTerminateViewModel(
+  insuranceId: String?,
+  getContractsToTerminateUseCase: GetContractsToTerminateUseCase,
+  terminateInsuranceRepository: TerminateInsuranceRepository,
+  featureManager: FeatureManager,
+) : MoleculeViewModel<ChooseInsuranceToTerminateEvent, ChooseInsuranceToTerminateStepUiState>(
+    initialState = ChooseInsuranceToTerminateStepUiState.Loading,
+    presenter = ChooseInsuranceToTerminatePresenter(
+      insuranceId = insuranceId,
+      getContractsToTerminateUseCase = getContractsToTerminateUseCase,
+      terminateInsuranceRepository = terminateInsuranceRepository,
+      featureManager = featureManager,
+    ),
+  )
+
+private class ChooseInsuranceToTerminatePresenter(
+  private val insuranceId: String?,
+  private val getContractsToTerminateUseCase: GetContractsToTerminateUseCase,
+  private val terminateInsuranceRepository: TerminateInsuranceRepository,
+  private val featureManager: FeatureManager,
+) : MoleculePresenter<ChooseInsuranceToTerminateEvent, ChooseInsuranceToTerminateStepUiState> {
+  @Composable
+  override fun MoleculePresenterScope<ChooseInsuranceToTerminateEvent>.present(
+    lastState: ChooseInsuranceToTerminateStepUiState,
+  ): ChooseInsuranceToTerminateStepUiState {
+    var loadIteration by remember { mutableIntStateOf(0) }
+    var currentSelectedId by remember { mutableStateOf(insuranceId) }
+    var terminationStep: TerminateInsuranceStep? by remember { mutableStateOf(null) }
+    var currentPartialState by remember {
+      mutableStateOf(lastState.toPartialState())
+    }
+
+    CollectEvents { event ->
+      when (event) {
+        ChooseInsuranceToTerminateEvent.RetryLoadData -> {
+          loadIteration++
+        }
+
+        is ChooseInsuranceToTerminateEvent.SelectInsurance -> {
+          currentSelectedId = event.insurance.id
+          if (currentPartialState is PartialUiState.Success) {
+            currentPartialState =
+              (currentPartialState as PartialUiState.Success).copy(selectedInsurance = event.insurance)
+          }
+        }
+      }
+    }
+
+    LaunchedEffect(currentSelectedId) {
+      val id = currentSelectedId
+      if (id != null) {
+        terminationStep = terminateInsuranceRepository.startTerminationFlow(InsuranceId(id)).getOrNull()
+        logcat { "mariia: LaunchedEffect(currentSelectedId) finished, terminationStep: $terminationStep" }
+      }
+    }
+
+    LaunchedEffect(loadIteration) {
+      currentPartialState = PartialUiState.Loading
+      combine(
+        flow { emit(getContractsToTerminateUseCase.invoke()) },
+        featureManager.isFeatureEnabled(Feature.TERMINATION_FLOW),
+      ) { contractsResult, isTerminationFlowEnabled ->
+        contractsResult to isTerminationFlowEnabled
+      }.collect { (contractsResult, isTerminationFlowEnabled) ->
+        logcat { "mariia: contractsResult: $contractsResult isTerminationFlowEnabled: $isTerminationFlowEnabled" }
```

Comment:
Remember to remove the log. Or if you think the log itself will be useful just remove your name but keep the log and give it a better "title" at the start of it.